### PR TITLE
Update lint configuration to use eslint

### PR DESCRIPTION
**Pull request by lucasecdb**
**Note: highly recommended that you review this PR commit by commit**

these changes are linked in [this](https://lucas--storecomponents.myvtex.com/shirt/Clothing?map=ft%2Cc) workspace.

8c6130b: removes tslint packages and config files and add eslint based on [this recipe](https://docs.google.com/document/d/1ePi5Mn3pe1dOorKrMtC-MclBDiIDimVzf0Uj61W8lA4/edit), with some additional disabled rules so we don't have to deal with them _right now_.

9c2e11e: with the linter change, some errors showed up regarding some unused variables, and a few other things. this commit addresses all **errors**, but some warnings were ignored (or disabled in the first commit)

aaf2923: this is just prettier's work, you can probably skip this.

### CODE REVIEW

File: react/tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,6 @@
     "noImplicitAny": true,
     "noImplicitReturns": true,
     "noImplicitThis": true,
-    "noUnusedLocals": true,
```
Comment (klzns) [inline react/tsconfig.json]: Why remove that?

Comment (lucasecdb) [reply react/tsconfig.json]: the `@typescript-eslint/no-unused-vars` already covers that

File: react/components/ExtensionPointComponent.tsx [hunk 2]
```diff
@@ -158,15 +159,15 @@ class ExtensionPointComponent extends PureComponent<
 
     // A children of this extension point throwed an uncaught error
     // Only show errors in production if the entire page explodes. (Ignore nested extension points)
-    if (error) {
+    if (error && errorInfo) {
```
Comment (brunoabreu) [inline react/components/ExtensionPointComponent.tsx]: `ExtensionPointError` accepts `errorInfo` as undefined.

```tsx
public render() {
    const {treePath, error, errorInfo} = this.props
    const {errorDetails} = this.state
    const componentStack = errorInfo && errorInfo.componentStack
```

Changing this `if` can change the current behavior.

File: react/components/MaybeContext.tsx [hunk 3]
```diff
@@ -8,23 +8,39 @@ interface Props {
...
-      ...pageContextProps,
-      nextTreePath: nestedPage,
-      params,
-      query,
-      ...context!.props
+    let contextComponent = context && context.component
+    let props
+
+    if (context) {
+      contextComponent = context.component
```
Comment (brunoabreu) [inline react/components/MaybeContext.tsx]: Why `contextComponent` is being assigned twice?

Comment (lucasecdb) [reply react/components/MaybeContext.tsx]: I removed the first assignment

**Pull request merged**